bench/irmin-pack/tree.ml

```diff
@@ -0,0 +1,351 @@
+  let config =
+    {
+      ncommits;
+      ncommits_log;
+      operations_file;
+      root = "test-bench";
+      mode;
+      depth;
+      width;
+      nchain_trees;
+      nlarge_trees;
+    }
+  in
+  let run_benchmarks () =
+    (match config.mode with
+    | `All -> [ `Tezos_log; `Chains; `Large ]
+    | (`Chains | `Large | `Tezos_log) as mode -> [ mode ])
+    |> Lwt_list.map_s (fun mode ->
+           init config.root;
+           if mode = `Tezos_log then (
+             Parse_log.prepare_log config.operations_file;
+             let n = Parse_log.populate_array "tezos.log" config.ncommits_log in
+             config.ncommits_log <- n;
+             Bench_inodes_32.run mode config >|= fun result_32 ->
+             Parse_log.cleanup ();
+             (mode, result_32, None))
+           else
+             Bench_inodes_32.run mode config >>= fun result_32 ->
+             Bench_inodes_2.run mode config >|= fun result_2 ->
+             (mode, result_32, Some result_2))
+  in
+  let results = Lwt_main.run (run_benchmarks ()) in
+  let pp_result fmt (mode, result_32, result_2) =
+    Format.fprintf fmt
+      "Configuration:@\n @[%a@]@\n@\nResults for inodes [32, 256]:@\n @[%a@]@\n"
+      (pp_config mode) config Benchmark.pp_results result_32;
+    match result_2 with
+    | None -> ()
+    | Some r ->
+        Format.fprintf fmt "Results for inodes [2, 5]:@\n @[%a@]@\n"
+          Benchmark.pp_results r
+  in
+  Fmt.pf Format.std_formatter "%a@."
+    Fmt.(list ~sep:(any "@\n@\n") pp_result)
+    results
+
+open Cmdliner
+
+let mode =
+  [
+    ("tezos-log", `Tezos_log);
+    ("chains", `Chains);
+    ("large", `Large);
+    ("all", `All);
+  ]
+
+let mode =
+  let doc = Arg.info ~doc:(Arg.doc_alts_enum mode) [ "mode" ] in
+  Arg.(value @@ opt (Arg.enum mode) `All doc)
+
+let ncommits =
+  let doc =
+    Arg.info ~doc:"Number of commits for the large and chain modes."
+      [ "n"; "ncommits" ]
+  in
+  Arg.(value @@ opt int 10 doc)
+
+let ncommits_log =
+  let doc =
+    Arg.info ~doc:"Number of commits to read from log." [ "ncommits_log" ]
+  in
+  Arg.(value @@ opt int 13315 doc)
+
+let depth =
+  let doc =
+    Arg.info ~doc:"Depth of a commit's tree in chains-mode." [ "d"; "depth" ]
+  in
+  Arg.(value @@ opt int 1000 doc)
+
+let nchain_trees =
+  let doc =
+    Arg.info ~doc:"Number of chain trees per commit in chains-mode."
+      [ "c"; "nchain" ]
+  in
+  Arg.(value @@ opt int 2 doc)
+
+let width =
+  let doc =
+    Arg.info ~doc:"Width of a commit's tree in large-mode." [ "w"; "width" ]
+  in
+  Arg.(value @@ opt int 1000000 doc)
+
+let nlarge_trees =
+  let doc =
+    Arg.info ~doc:"Number of large trees per commit in large-mode."
+      [ "l"; "nlarge" ]
+  in
+  Arg.(value @@ opt int 1 doc)
+
+let operations_file =
+  let doc =
+    Arg.info ~doc:"Compressed file containing the tree operations."
+      [ "f"; "file" ]
+  in
+  Arg.(value @@ opt string "bench/irmin-pack/tezos_log.tar.gz" doc)
```

Comment:
Are we sure we want to introduce a 52MB file in the repo (`tezos_log.tar.gz`)? I personally don't mind, but there are other ways to proceed.